src/pages/tutorials/build-an-nft-marketplace/index.md

```diff
@@ -431,542 +444,547 @@ This section guides you step-by-step in setting up an intuitive frontend.
-```
+    ```bash
+    cp -r ../reactboilerplateapp/ ./app
+    ```
 
-> Note : if you want to understand how it has been made from scratch look at [this training](https://github.com/marigold-dev/training-dapp-1#construction_worker-dapp)
+    > Note: if you want to understand how it has been made from scratch look at [this training](https://github.com/marigold-dev/training-dapp-1#construction_worker-dapp)
 
-It is easier on frontend side to use typed objects. Taqueria provides a plugin to generate Typescript classes from your Michelson code.
+    It is easier on frontend side to use typed objects. Taqueria provides a plugin to generate Typescript classes from your Michelson code.
 
-Install the plugin, then generate a representation of your smart contract objects that writes these files to your frontend app source code.
+    Install the plugin, then generate a representation of your smart contract objects that writes these files to your frontend app source code.
 
-Finally, run the server
+2. Run the server
 
-```bash
-taq install @taqueria/plugin-contract-types
-taq generate types ./app/src
-cd app
-yarn install
-yarn dev
-```
+    ```bash
+    taq install @taqueria/plugin-contract-types
+    taq generate types ./app/src
+    cd app
+    yarn install
+    yarn dev
+    ```
 
-> Note : On a **Mac** :green_apple:, sometimes `sed` commands do not work exactly the same as Unix commands. Look at the start script on package.json for Mac below :
-> `   "dev": "if test -f .env; then sed -i '' \"s/\\(VITE_CONTRACT_ADDRESS *= *\\).*/\\1$(jq -r 'last(.tasks[]).output[0].address' ../.taq/testing-state.json)/\" .env ; else jq -r '\"VITE_CONTRACT_ADDRESS=\" + last(.tasks[]).output[0].address' ../.taq/testing-state.json > .env ; fi && vite",`
+    > Note: On a **Mac**:green_apple:, sometimes `sed` commands do not work exactly the same as Unix commands. Look at the start script on package.json for Mac below:
+    > `   "dev": "if test -f .env; then sed -i '' \"s/\\(VITE_CONTRACT_ADDRESS *= *\\).*/\\1$(jq -r 'last(.tasks[]).output[0].address' ../.taq/testing-state.json)/\" .env ; else jq -r '\"VITE_CONTRACT_ADDRESS=\" + last(.tasks[]).output[0].address' ../.taq/testing-state.json > .env ; fi && vite",`
 
-The website is ready! You have:
+    The website is ready! You have:
 
-- last deployed contract address always is refreshed from **taqueria** configuration at each start
-- login/logout
-- the general layout / navigation
+    - last deployed contract address always is refreshed from **taqueria** configuration at each start
+    - login/logout
+    - the general layout / navigation
 
-If you try to connect you are redirected to `/` path that is also the wine catalog.
+    If you try to connect you are redirected to `/` path that is also the wine catalog.
 
-There are no bottle collections yet, so you have to create the mint page.
+    There are no bottle collections yet, so you have to create the mint page.
 
 ### Step 2: Mint Page
```

Comment:
When I copy the following code, the editor complains about it. Is there any possible to provide the complete Mintpage.tsx code?